tests/integration/test_stepfunctions.py

```diff
@@ -499,6 +500,42 @@ def check_invocations():
         cleanup(sm_arn, state_machines_before, stepfunctions_client)
         events.delete_event_bus(Name=bus_name)
 
+    def test_create_state_machines_in_parallel(self, stepfunctions_client):
+        """
+        Perform a test that creates a series of state machines in parallel. Without concurrency control, using
+        StepFunctions-Local, the following error is pretty consistently reproducible:
+
+        botocore.errorfactory.InvalidDefinition: An error occurred (InvalidDefinition) when calling the
+        CreateStateMachine operation: Invalid State Machine Definition: ''DUPLICATE_STATE_NAME: Duplicate State name:
+        MissingValue at /States/MissingValue', 'DUPLICATE_STATE_NAME: Duplicate State name: Add at /States/Add''
+        """
+        role_arn = arns.role_arn("sfn_role")
+        definition = clone(STATE_MACHINE_CHOICE)
+        lambda_arn_4 = arns.lambda_function_arn(TEST_LAMBDA_NAME_4)
+        definition["States"]["Add"]["Resource"] = lambda_arn_4
+        definition = json.dumps(definition)
+        results = []
+
+        def _create_sm(*_):
+            sm_name = f"sm-{short_uid()}"
+            result = stepfunctions_client.create_state_machine(
+                name=sm_name, definition=definition, roleArn=role_arn
+            )
+            assert result["ResponseMetadata"]["HTTPStatusCode"] == 200
+            results.append(result)
+            stepfunctions_client.describe_state_machine(stateMachineArn=result["stateMachineArn"])
+            stepfunctions_client.list_tags_for_resource(resourceArn=result["stateMachineArn"])
+
+        try:
+            num_machines = 30
+            parallelize(_create_sm, list(range(num_machines)), size=2)
+            assert len(results) == num_machines
+        finally:
+            for machine in results:
+                stepfunctions_client.delete_state_machine(
+                    stateMachineArn=machine["stateMachineArn"]
+                )
```

Comment:
```suggestion
        def _create_sm(*_):
            sm_name = f"sm-{short_uid()}"
            result = stepfunctions_client.create_state_machine(
                name=sm_name, definition=definition, roleArn=role_arn
            )
            assert result["ResponseMetadata"]["HTTPStatusCode"] == 200
            cleanups.append(lambda: stepfunctions_client.delete_state_machine(stateMachineArn=result["stateMachineArn"]))
            results.append(result)
            stepfunctions_client.describe_state_machine(stateMachineArn=result["stateMachineArn"])
            stepfunctions_client.list_tags_for_resource(resourceArn=result["stateMachineArn"])

        num_machines = 30
        parallelize(_create_sm, list(range(num_machines)), size=2)
        assert len(results) == num_machines
```

Reply:
cleanup-nit: a bit more robust when dealing with multiple cleanups and avoids the try/finally block